### PR TITLE
Interop examples created by Gary Bilkus to supplement existing docume…

### PR DESCRIPTION
**Pull request by bilkusg**
…ntation

### CODE REVIEW

**Comment by MangelMaxime** [comment]
I will update a little your code to have a better display and more user friendly.

I will ping you when sent, and feel free to tell me if my proposition are ok for you or no :)

**Comment by MangelMaxime** [comment]
Ok I took the time to read your code, and I don't think that using the `Utils.c` function is a good way to comment the code.

Problem is the output is really compact (I tried to fix that but wasn't feeling that nice), lack color, etc.

I personnaly would prefer, if the F# code is commented the normal way. Because, there we can indent the code, have synthax coloration support, tooltip, etc.

Showing the console ouput in the preview page, is nice if we only output minimal info.

For example, I would rewrite this lines:

```
   c "START HERE"
   c "Manipulating a value which is really a pojo object under the hood"
   c "Add a new value to an object"
   c "boringObj.newValue = \"newValue\""

   boring1Obj?newValue <- "newValue"
   Browser.console.log("boring = ",boring1Obj)
```

as

```fs
   // Manipulating a value which is really a pojo object under the hood
   // Add a new value to an object
   // boringObj.newValue = "newValue"
   boring1Obj?newValue <- "newValue"
   Utils.code "boring1Obj?newValue <- \"newValue\"" boring1Obj
```

with:

```fs
module Utils =
    open Fable.Core
    open Fable.Import

    [<Emit("JSON.stringify($0, null, 4) + ''")>]
    let anyToString (_: obj) : string= jsNative

    let code codeString value = 
        Browser.console.log("> " + codeString + ":")
        Browser.console.log(anyToString value)
```

This result in:

capture d ecran 2018-03-08 a 14 56 12

File: public/samples/interop-samples/interop-examples.fs
Comment (MangelMaxime) [inline public/samples/interop-samples/interop-examples.fs]: You need to split the lines because it's not easy to read in the browser

File: public/samples/interop-samples/interop-examples.fs
Comment (MangelMaxime) [inline public/samples/interop-samples/interop-examples.fs]: Please use "4 spaces indentation or tab" here you are usinfg 1 space to indent the code and it's really hard to read

File: src/App/Widgets/samples.json
Comment (MangelMaxime) [inline src/App/Widgets/samples.json]: The two lines are wrong.
Should be:

```fs
"fsharpCode": "samples/interop-samples/interop-samples.fs",
"htmlCode": "samples/interop-samples/interop-samples.html"
```

**Review by MangelMaxime — CHANGES_REQUESTED** [review]

File: public/samples/interop-samples/interop-examples.html
Comment (MangelMaxime) [inline public/samples/interop-samples/interop-examples.html]: I would write the explanation text here and with something like:

```fs
  <h2>
    The section below contains the console output from running the program.
  </h2>
  <h4>What's most interesting is the generated code, so switch to that tab and compare with the original program.
  </h4>
  <p>Version 1.0 March 2018</p>
```

It's less agressive :)

**Review by MangelMaxime — CHANGES_REQUESTED** [review]

**Comment by MangelMaxime** [comment]
The content is really interesting 👍 , we need to work on the visual aspect :)

File: public/samples/interop-samples/interop-examples.fs
Comment (MangelMaxime) [inline public/samples/interop-samples/interop-examples.fs]: `javascript` should be written `JavaScript` this occured several times in the file

**Comment by bilkusg** [comment]
Hi,
Thanks for your comments and kind words on the substance. 
I will work on incorporating the requested updates and let you know when I've changed it.

**Comment by bilkusg** [comment]
Just one point - I can't find a way to move the console output div below the text rather than above it, as you suggest. This is because the javascript to copy the console can't see inside the iframe so it creates a new window above it..... Hope that's OK. 

I'm about half way through changing the notation style now.

**Comment by bilkusg** [comment]
OK I have updated the repo and pull request with the requested changes. Let me know if this is now what you want.

**Review by MangelMaxime — APPROVED** [review]

**Comment by MangelMaxime** [comment]
@alfonsogarciacaro Can I let you review the changes and decide to merge or no the PR.

I made a pass on the code to made it cleaner. Still not sure of the result.

File: public/samples/interop-samples/interop-samples.fs
Comment (alfonsogarciacaro) [inline public/samples/interop-samples/interop-samples.fs:239]: An alternative syntax is:

```fsharp
let [<Global("boringFunction")>] boring1Function(x: int) (y: int): string = jsNative
```

```fsharp
let [<Global("boringFunction")>] boring1Function(x: int, y: int): string = jsNative
```

File: public/samples/interop-samples/interop-samples.fs
Comment (alfonsogarciacaro) [inline public/samples/interop-samples/interop-samples.fs:66]: @bilkusg Sorry, forgot to tell you that when importing a global function you can also type it as `Fable.Core.JsInterop.JsFunc`. Then you can invoke it dynamically later:

```fsharp
boring1String.Invoke(1, 2, "foo", "bar")
```

**Review by alfonsogarciacaro — APPROVED** [review]

**Comment by alfonsogarciacaro** [comment]
@bilkusg @MangelMaxime Sorry, I thought you were dealing with this and didn't check the last notification 😅 I just left a couple of minor comments but this is OK for me to merge :+1:

**Comment by bilkusg** [comment]
Is there anything else needed before this can go live? I'm hoping to refer to the repl in some articles I'm writing, but I obviously need to wait until the merge happens.

Thanks,

**Comment by MangelMaxime** [comment]
@bilkusg Just need to change the code according to Alfonso comments and then we are ready to merge.

Not for myself, make sure to use Fable 1.3.7 before depoying new version because @ncave suppose there is a bug in the REPL with newer version of Fable .

**Comment by alfonsogarciacaro** [comment]
I'd say let's merge this. Comments are not so important anyways :smile: I'll push the sample :+1:

**Comment by MangelMaxime** [comment]
Ok thanks @alfonsogarciacaro

**Pull request merged**
